testing/test-nx-utils/src/lib/utils/tree.unit.test.ts

```diff
@@ -0,0 +1,31 @@
+import { join } from 'node:path';
+import { createTreeWithEmptyWorkspace } from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
+import { describe, expect } from 'vitest';
+import { readJsonFile } from '@code-pushup/utils';
+import { materializeTree } from './tree';
+
+describe('materializeTree', () => {
+  const baseDir = join('tmp', 'materialize-tree');
```

Comment:
Since this is interacting with the actual file system, I would make it an integration test.

Alternatively, you could use `memfs` and keep this as a unit test.

Reply:
Very true. I on purpose did not use memfs. so will rename to integration test

Reply:
done